master/internal/core_experiment.go

```diff
@@ -83,8 +83,46 @@ func (m *Master) getExperimentCheckpointsToGC(c echo.Context) (interface{}, erro
 	if err := api.BindArgs(&args, c); err != nil {
 		return nil, err
 	}
-	return m.db.ExperimentCheckpointsToGCRaw(
+
+	checkpointUUIDs, err := m.db.ExperimentCheckpointsToGCRaw(
 		args.ExperimentID, args.ExperimentBest, args.TrialBest, args.TrialLatest)
+	if err != nil {
+		return nil, err
+	}
+	checkpointsDB, err := m.db.CheckpointByUUIDs(checkpointUUIDs)
+	if err != nil {
+		return nil, err
+	}
+
+	expConfig, err := m.db.ExperimentConfig(args.ExperimentID)
+	if err != nil {
+		return nil, err
+	}
+	var checkpoints []map[string]interface{}
+	for _, cDB := range checkpointsDB {
```

Comment:
i would just return a more sane type here that contains all the data (maybe just return the model.Checkpoint) and then do this logic on the cli side